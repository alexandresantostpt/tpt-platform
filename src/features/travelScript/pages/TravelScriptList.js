import { events } from '@constants/events'

import PropTypes from 'prop-types'
import PubSub from 'pubsub-js'
import React, { useEffect } from 'react'
import styled from 'styled-components'

import { colors } from '@helpers/colors'

import { i18n } from '@i18n'

import Button from '@components/button/Button'
import If from '@components/if/If'

import { not } from '@utils/functions'

import { Route, Router, Switch } from 'react-router'
import { withRouter } from 'react-router-dom'

import App from '@containers/App'
import Container from '@containers/Container'
import Summary from '@containers/Summary'

import { history } from '@cfg/history'
import { travelServiceRoutes } from '@cfg/routes'

import TravelScriptDay from '../components/day/TravelScriptDay'
import AddCollaborators from '../components/addCollaborators/AddCollaborators'

import TravelScriptModel from '../models/TravelScriptModel'
import UserModel from '@/features/login/UserModel'

const TravelScriptHeader = styled.div`
    display: flex;
    justify-content: space-between;
    padding: 0 52px 59px;
`
const TravelMaterialButton = styled.div`
    display: flex;
    justify-content: flex-end;
    padding: 0 52px 59px;
`

const TravelScriptList = ({
    match,
    requestCloseMaterialAction,
    requestDelServiceAction,
    requestGetSingleAction,
    requestUnshareAction,
    requestUpdateCityAction,
    requestUpdateMaterialAction,
    requestShareTravelAction,
    travelScript,
    users
}) => {
    const {
        params: { id }
    } = match

    useEffect(() => {
        requestGetSingleAction({ id })
    }, [])

    useEffect(() => {
        PubSub.subscribe(events.SERVICE_DELETED, () => requestGetSingleAction({ id }))
        return () => PubSub.unsubscribe(events.SERVICE_DELETED)
    }, [])

    const addCollaborator = email => requestShareTravelAction({ email, id })

    const closeMaterial = () => {
        if (id) {
            requestCloseMaterialAction(id)
        }
    }

    const removeCollaborator = collaborator => requestUnshareAction({ travelId: id, userId: collaborator.id })

    const updateMaterial = () => {
        if (id) {
            requestUpdateMaterialAction(id)
        }
    }

    return (
        <App travelScript={travelScript}>
            <Container>
                {travelScript !== null && (
                    <>
                        <TravelScriptHeader>
                            <div>{travelScript.travel.title}</div>
                            <AddCollaborators
                                addCollaborator={addCollaborator}
                                collaborators={users}
                                removeCollaborator={removeCollaborator}
                            />
                        </TravelScriptHeader>
                        <TravelMaterialButton>
                            <If condition={travelScript !== null && not(travelScript.travel.closed) && travelScript.hasServices()}>
                                <Button
                                    color={colors.purple}
                                    disabled={false}
                                    fill={true}
                                    onClick={closeMaterial}
                                    text={i18n.t('buttons.closeMaterial')}
                                />
                            </If>
                        </TravelMaterialButton>
                        <TravelMaterialButton>
                            <If condition={travelScript.travel.closed && travelScript.edited}>
                                <Button
                                    color={colors.purple}
                                    disabled={false}
                                    fill={true}
                                    onClick={updateMaterial}
                                    text={i18n.t('buttons.updateMaterial')}
                                />
                            </If>
                        </TravelMaterialButton>
                        {travelScript.dates.map((item, index) => (
                            <TravelScriptDay
                                cities={travelScript.travel.citiesDestiny}
                                date={item.date}
                                dateId={item._id}
                                dayNumber={++index}
                                delService={requestDelServiceAction}
                                id={id}
                                key={index}
                                onCityChange={option =>
                                    requestUpdateCityAction({
                                        city: option.obj,
                                        dateId: item._id,
                                        scriptId: travelScript.id
                                    })
                                }
                                scriptId={travelScript.id}
                                selectedCity={item.city}
                                services={item.services}
                            />
                        ))}
                    </>
                )}
            </Container>
            <Summary>
                <Router history={history}>
                    <Switch>
                        {travelServiceRoutes.map((route, index) => (
                            <Route {...route} exact key={route.name || index} />
                        ))}
                    </Switch>
                </Router>
            </Summary>
        </App>
    )
}

TravelScriptList.defaultProps = {
    travelScript: null,
    users: []
}

TravelScriptList.propTypes = {
    match: PropTypes.object.isRequired,
    requestCloseMaterialAction: PropTypes.func.isRequired,
    requestDelServiceAction: PropTypes.func.isRequired,
    requestGetSingleAction: PropTypes.func.isRequired,
    requestShareTravelAction: PropTypes.func.isRequired,
    requestUnshareAction: PropTypes.func.isRequired,
    requestUpdateCityAction: PropTypes.func.isRequired,
    requestUpdateMaterialAction: PropTypes.func.isRequired,
    travelScript: PropTypes.objectOf(TravelScriptModel),
    users: PropTypes.arrayOf(UserModel)
}

export default withRouter(TravelScriptList)
