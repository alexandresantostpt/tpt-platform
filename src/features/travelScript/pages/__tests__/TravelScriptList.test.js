import configureMockStore from 'redux-mock-store'
import moment from 'moment'
import React from 'react'
import renderer from 'react-test-renderer'

import { List, Map, Record } from 'immutable'
import { Provider } from 'react-redux'

import TravelModel from '@features/travel/TravelModel'
import TravelScriptModel from '../../models/TravelScriptModel'

import TravelScriptList from '../TravelScriptList'

describe('Tests for TravelScript component', () => {
    let mockStore = {}
    let store = {}
    let toast = {}
    let task = {}

    const travel = new TravelModel({
        __v: 0,
        _id: '5c52ebcd5fc7c9b1433ab704',
        citiesDestiny: ['Campinas', 'SÃ£o Paulo'],
        clientCpf: '123.123.123-12',
        clientEmail: 'email@gmail.com',
        clientName: 'TPT EDIT',
        clients: [
            {
                __v: 0,
                cpf: '123.123.123-12',
                createdAt: '2019-03-01T12:35:34.965Z',
                email: 'email@gmail.com',
                id: '5c792716e6e6030f80b85732',
                name: 'TPT EDIT',
                updatedAt: '2019-03-01T12:35:34.965Z'
            }
        ],
        closed: false,
        createdAt: '2019-01-31T12:36:29.463Z',
        travelEndDate: '2019-01-30T12:10:10.000Z',
        travelStartDate: '2019-02-05T17:10:10.000Z',
        updatedAt: '2019-01-31T12:36:29.463Z'
    })

    const service = {
        service: {
            __v: 1,
            address: 'Teste DOIS',
            clients: ['5c792716e6e6030f80b85732'],
            confirmedBy: 'Teste',
            createdAt: '2019-03-07T21:52:59.347Z',
            dressCode: 'Casual',
            id: '5c8192bb650a39eb9dd49249',
            library: {
                name: 'Teste'
            },
            peopleCount: 1,
            reserveDate: '2019-02-01T00:00:00.000Z',
            reserveHour: '2019-02-01T02:30:00.000Z',
            reserveNumber: 123,
            script: '5c819219650a3965a5d49243',
            scriptDate: '2019-02-01T00:00:00.000Z',
            type: 'RESTAURANT',
            updatedAt: '2019-03-08T12:38:21.680Z'
        }
    }

    const date = {
        city: 'Campinas',
        date: moment('15-02-2019', 'DD-MM-YYYY').utc(false),
        services: [service]
    }

    const travelScript = new TravelScriptModel({
        __v: '1.0',
        _id: '1',
        createdAt: '2019-02-21',
        dates: [date],
        services: [],
        travel,
        updatedAt: '2019-02-21'
    })

    beforeEach(() => {
        task = {
            __v: 28,
            _id: '5c9e31f8c57b4917b426e334',
            createdAt: '2019-03-29T14:55:52.764Z',
            describe: 'teste 123',
            done: false,
            updatedAt: '2019-03-29T14:55:52.764Z',
            user: '5c9dfb65d1cf42bd5fa66b3d'
        }
        toast = { icon: 'check', message: 'success', type: 'success' }

        mockStore = configureMockStore()
        store = mockStore(
            Record({
                tasksReducers: Map({
                    list: List.of(task)
                }),
                toastReducers: Map({
                    list: List.of(toast)
                })
            })()
        )
    })

    it('Should render withou crash', () => {
        const match = {
            params: {
                id: '5c52ebcd5fc7c9b1433ab704'
            }
        }
        const tree = renderer
            .create(
                <Provider store={store}>
                    <TravelScriptList
                        match={match}
                        requestGetSingleAction={jest.fn()}
                        requestUpdateCityAction={jest.fn()}
                        travelScript={travelScript}
                    />
                </Provider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('Should render a closed travel', () => {
        const match = {
            params: {
                id: '5c52ebcd5fc7c9b1433ab704'
            }
        }

        const closeTravelScript = new TravelScriptModel({
            __v: '1.0',
            _id: '1',
            createdAt: '2019-02-21',
            dates: [date],
            travel: { ...travel, closed: true },
            updatedAt: '2019-02-21'
        })
        const tree = renderer
            .create(
                <Provider store={store}>
                    <TravelScriptList
                        match={match}
                        requestGetSingleAction={jest.fn()}
                        requestUpdateCityAction={jest.fn()}
                        travelScript={closeTravelScript}
                    />
                </Provider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
