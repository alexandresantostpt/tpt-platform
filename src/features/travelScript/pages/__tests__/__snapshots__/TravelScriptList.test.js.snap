// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for TravelScript component Should render a closed travel 1`] = `
Array [
  <header
    className="Header__StyledHeader-od5ehu-0 hIVEkQ"
  >
    <span
      className="Brand__StyledBrand-sc-19fy8m0-0 jmSzTQ"
    />
    <nav
      className="Menu__StyledMenu-w0f985-0 kGkgdR"
    >
      <ul
        className="MenuList__StyledMenuList-sc-1bxu43p-0 ixFBrS"
      >
        <li
          className="MenuItem__StyledMenuItem-zqooc6-0 UmzxF"
        >
          <span
            className="icon icon-plane Icon__StyledIcon-crr1cg-0 dYwZuZ"
            disabled={false}
            fontSize="2.5rem"
            onClick={null}
          />
          <a
            className="MenuLink__StyledMenuLink-sc-14ijefa-0 fyvrdK"
            to="/"
          >
            Viagens
          </a>
        </li>
        <li
          className="MenuItem__StyledMenuItem-zqooc6-0 UmzxF"
        >
          <span
            className="icon icon-clients Icon__StyledIcon-crr1cg-0 bEFLZZ"
            disabled={false}
            fontSize="2rem"
            onClick={null}
          />
          <a
            className="MenuLink__StyledMenuLink-sc-14ijefa-0 fyvrdK"
            to="/client"
          >
            Clientes
          </a>
        </li>
        <li
          className="MenuItem__StyledMenuItem-zqooc6-0 UmzxF"
        >
          <span
            className="icon icon-bookmark Icon__StyledIcon-crr1cg-0 jtTFhN"
            disabled={false}
            fontSize="2.3rem"
            onClick={null}
          />
          <a
            className="MenuLink__StyledMenuLink-sc-14ijefa-0 fyvrdK"
            to="/library"
          >
            Biblioteca
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="Header__StyledOptions-od5ehu-1 hmosuQ"
    >
      <div
        className="Dropdown__StyledDropdown-sc-14kwj8x-0 iZGQRX"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <img
          alt="Johnny Silva"
          className="DropdownPicture__StyledDropdownPicture-ylb9qy-0 iHpJFt"
          src="https://api.adorable.io/avatars/100/tpt-platform"
        />
        <ul
          className="DropdownList__StyledDropdownList-vjd4uo-0 iDpmEs"
        >
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 jhIdjc"
            onClick={null}
          >
            <span
              className="DropdownTitle__StyledDropdownTitle-y9o5oh-0 hFdNKe"
            >
              Meu perfil
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Registro de atividades
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              Atualizado às 8:45
            </span>
          </li>
        </ul>
      </div>
      <div
        className="Dropdown__StyledDropdown-sc-14kwj8x-0 iZGQRX"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="icon icon-notification Icon__StyledIcon-crr1cg-0 llBDkg"
          disabled={false}
          fontSize="1.75rem"
          onClick={null}
        />
        <ul
          className="DropdownList__StyledDropdownList-vjd4uo-0 iDpmEs"
        >
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 jhIdjc"
            onClick={null}
          >
            <span
              className="DropdownTitle__StyledDropdownTitle-y9o5oh-0 hFdNKe"
            >
              Notificações
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Alteração serviço
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              Há 2 minutos
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Lembrete: Preencher informações
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              Ontem às 15:30
            </span>
          </li>
        </ul>
      </div>
      <div
        className="Dropdown__StyledDropdown-sc-14kwj8x-0 iZGQRX"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="icon icon-config Icon__StyledIcon-crr1cg-0 llBDkg"
          disabled={false}
          fontSize="1.75rem"
          onClick={null}
        />
        <ul
          className="DropdownList__StyledDropdownList-vjd4uo-0 iDpmEs"
        >
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 jhIdjc"
            onClick={null}
          >
            <span
              className="DropdownTitle__StyledDropdownTitle-y9o5oh-0 hFdNKe"
            >
              Configurações
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={[Function]}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Alterar senha
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 hOZFkb"
            onClick={[Function]}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Sair
            </span>
          </li>
        </ul>
      </div>
    </div>
  </header>,
  <section
    className="App__SideBar-h8qnv1-0 cfDVBS"
  >
    <h1>
      TravelCalendar
    </h1>
    <div
      className="Tasks__StyledTasks-sc-1hy8ads-0 cjVtaH"
    >
      <h2
        className="Title__StyledTitle-sc-11bxhsf-0 frSblw"
      >
        Tarefas
      </h2>
      <button
        className="AddButton__StyledAddButon-ppf0uf-0 gwSZMh"
        onClick={[Function]}
      >
        <span
          className="icon icon-plus Icon__StyledIcon-crr1cg-0 jfRxiR"
          disabled={false}
          fontSize="0.8rem"
          onClick={null}
        />
        Adicionar Tarefa
      </button>
      <div
        className="Task__StyledTask-sc-1fi2258-0 cEDnPQ"
      >
        <span
          className="TaskCheck__StyledTaskCheck-sc-1tgftv-0 kYiGLh"
          onClick={[Function]}
        >
          <button
            className="TaskCheck__StyledButton-sc-1tgftv-1 elQAyk"
          />
          <span
            className="icon icon-check Icon__StyledIcon-crr1cg-0 llBDkg"
            disabled={false}
            fontSize="1.75rem"
            onClick={null}
          />
        </span>
        <div
          className="TaskBody__StyledBody-sc-1p8y2k3-0 cOetdt"
        >
          <textarea
            autoComplete="off"
            className="TaskEdit__StyledTaskEdit-f5yune-0 kXgwgR"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            value="teste 123"
          />
        </div>
        <span
          className="TaskActions__StyledTaskActions-sc-92wkgs-0 fGXkbW"
        >
          <span
            className="icon icon-times Icon__StyledIcon-crr1cg-0 efWoYG"
            disabled={false}
            fontSize="1.5rem"
            onClick={[Function]}
          />
        </span>
      </div>
    </div>
  </section>,
  <main
    className="Container__StyledContainer-sc-1i8buc6-0 SsRsD"
  >
    <div
      className="TravelScriptList__TravelScriptHeader-ty58ph-0 fWwphJ"
    >
      <div />
      <div
        className="AddCollaborators__AddCollaboratorsContainer-sc-9j6qbx-0 jhjoiw"
      >
        <button
          className="StyledInviteButton-sc-1iige8l-0 cJgVMw"
          onClick={[Function]}
        >
          Convidar
        </button>
      </div>
    </div>
    <div
      className="TravelScriptList__TravelMaterialButton-ty58ph-1 gldxts"
    />
    <div
      className="TravelScriptList__TravelMaterialButton-ty58ph-1 gldxts"
    />
    <div
      className="TravelScriptDay__StyledTravelScriptDay-uatujh-0 faduyg"
    >
      <div
        className="TravelScriptDay__StyledSelects-uatujh-1 dFLRG"
      >
        <div
          className="CitySelect__StyledCitySelect-p43eaz-1 jykjhD"
        >
          <div
            className="CitySelect__StyledDayNumber-p43eaz-0 dnawfo"
          >
            1
          </div>
          <div
            className="Select__StyledSelect-sc-12397s2-0 eBEwLw"
          >
            <div
              className="SelectContainer__StyledSelectContainer-an3t0v-0 jYoLEQ"
            >
              <span
                className="SelectText-eq3jk4-0 iwqxNL"
                onClick={[Function]}
              >
                Selecione uma cidade...
              </span>
              <button
                className="SelectDropdown__StyledSelectDropdown-sc-12m970x-0 RvUCy"
                onClick={[Function]}
                type="button"
              />
            </div>
            <ul
              className="SelectList__StyledSelectList-lfsn73-0 hIsMxM"
            />
          </div>
          <div
            className="CitySelect__StyledLine-p43eaz-2 glAxln"
          />
        </div>
        <div
          className="ServiceSelect__StyledServiceSelect-sc-1rmby0e-0 kwWXAN"
        >
          <div
            className="Select__StyledSelect-sc-12397s2-0 eBEwLw"
          >
            <div
              className="SelectContainer__StyledSelectContainer-an3t0v-0 jYoLEQ"
            >
              <span
                className="SelectText-eq3jk4-0 iwqxNL"
                onClick={[Function]}
              >
                <span
                  className="icon icon-plus Icon__StyledIcon-crr1cg-0 llBDkg"
                  disabled={false}
                  fontSize="1.75rem"
                  onClick={null}
                />
              </span>
              <button
                className="SelectDropdown__StyledSelectDropdown-sc-12m970x-0 RvUCy"
                onClick={[Function]}
                type="button"
              />
            </div>
            <ul
              className="SelectList__StyledSelectList-lfsn73-0 hIsMxM"
            />
          </div>
        </div>
      </div>
      <div
        className="TravelScriptDay__StyledDate-uatujh-2 kemUYJ"
      >
        15/02
      </div>
      <div
        className="TravelScriptService__StyledTravelScript-n3v2ak-1 lhJYVv"
      >
        <span
          className="TravelScriptService__StyledDelIcon-n3v2ak-0 ciWpmh"
        >
          <span
            className="icon icon-times Icon__StyledIcon-crr1cg-0 efWoYG"
            disabled={false}
            fontSize="1.5rem"
            onClick={[Function]}
          />
        </span>
        <div
          className="TravelScriptService__StyledService-n3v2ak-2 cdILFI"
          onClick={[Function]}
        >
          <div
            className="TravelScriptService__StyledBullet-n3v2ak-3 bvzinQ"
            type="RESTAURANT"
          />
          <div
            className="TravelScriptService__StyledServiceInformation-n3v2ak-4 gaWVbb"
          >
            <div
              className="TravelScriptService__StyledType-n3v2ak-6 hGjoUD"
            >
              Restaurante
            </div>
            <span
              className="TravelScriptService__StyledName-n3v2ak-5 kRnSnh"
            >
              Teste
            </span>
          </div>
        </div>
      </div>
    </div>
  </main>,
  <section
    className="Summary__StyledSummary-sc-1mglm3x-0 fKeRKj"
  />,
]
`;

exports[`Tests for TravelScript component Should render withou crash 1`] = `
Array [
  <header
    className="Header__StyledHeader-od5ehu-0 hIVEkQ"
  >
    <span
      className="Brand__StyledBrand-sc-19fy8m0-0 jmSzTQ"
    />
    <nav
      className="Menu__StyledMenu-w0f985-0 kGkgdR"
    >
      <ul
        className="MenuList__StyledMenuList-sc-1bxu43p-0 ixFBrS"
      >
        <li
          className="MenuItem__StyledMenuItem-zqooc6-0 UmzxF"
        >
          <span
            className="icon icon-plane Icon__StyledIcon-crr1cg-0 dYwZuZ"
            disabled={false}
            fontSize="2.5rem"
            onClick={null}
          />
          <a
            className="MenuLink__StyledMenuLink-sc-14ijefa-0 fyvrdK"
            to="/"
          >
            Viagens
          </a>
        </li>
        <li
          className="MenuItem__StyledMenuItem-zqooc6-0 UmzxF"
        >
          <span
            className="icon icon-clients Icon__StyledIcon-crr1cg-0 bEFLZZ"
            disabled={false}
            fontSize="2rem"
            onClick={null}
          />
          <a
            className="MenuLink__StyledMenuLink-sc-14ijefa-0 fyvrdK"
            to="/client"
          >
            Clientes
          </a>
        </li>
        <li
          className="MenuItem__StyledMenuItem-zqooc6-0 UmzxF"
        >
          <span
            className="icon icon-bookmark Icon__StyledIcon-crr1cg-0 jtTFhN"
            disabled={false}
            fontSize="2.3rem"
            onClick={null}
          />
          <a
            className="MenuLink__StyledMenuLink-sc-14ijefa-0 fyvrdK"
            to="/library"
          >
            Biblioteca
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="Header__StyledOptions-od5ehu-1 hmosuQ"
    >
      <div
        className="Dropdown__StyledDropdown-sc-14kwj8x-0 iZGQRX"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <img
          alt="Johnny Silva"
          className="DropdownPicture__StyledDropdownPicture-ylb9qy-0 iHpJFt"
          src="https://api.adorable.io/avatars/100/tpt-platform"
        />
        <ul
          className="DropdownList__StyledDropdownList-vjd4uo-0 iDpmEs"
        >
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 jhIdjc"
            onClick={null}
          >
            <span
              className="DropdownTitle__StyledDropdownTitle-y9o5oh-0 hFdNKe"
            >
              Meu perfil
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Registro de atividades
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              Atualizado às 8:45
            </span>
          </li>
        </ul>
      </div>
      <div
        className="Dropdown__StyledDropdown-sc-14kwj8x-0 iZGQRX"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="icon icon-notification Icon__StyledIcon-crr1cg-0 llBDkg"
          disabled={false}
          fontSize="1.75rem"
          onClick={null}
        />
        <ul
          className="DropdownList__StyledDropdownList-vjd4uo-0 iDpmEs"
        >
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 jhIdjc"
            onClick={null}
          >
            <span
              className="DropdownTitle__StyledDropdownTitle-y9o5oh-0 hFdNKe"
            >
              Notificações
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Alteração serviço
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              Há 2 minutos
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={null}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Lembrete: Preencher informações
            </span>
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
            >
              Ontem às 15:30
            </span>
          </li>
        </ul>
      </div>
      <div
        className="Dropdown__StyledDropdown-sc-14kwj8x-0 iZGQRX"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="icon icon-config Icon__StyledIcon-crr1cg-0 llBDkg"
          disabled={false}
          fontSize="1.75rem"
          onClick={null}
        />
        <ul
          className="DropdownList__StyledDropdownList-vjd4uo-0 iDpmEs"
        >
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 jhIdjc"
            onClick={null}
          >
            <span
              className="DropdownTitle__StyledDropdownTitle-y9o5oh-0 hFdNKe"
            >
              Configurações
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
            onClick={[Function]}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Alterar senha
            </span>
          </li>
          <li
            className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 hOZFkb"
            onClick={[Function]}
          >
            <span
              className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
            >
              Sair
            </span>
          </li>
        </ul>
      </div>
    </div>
  </header>,
  <section
    className="App__SideBar-h8qnv1-0 cfDVBS"
  >
    <h1>
      TravelCalendar
    </h1>
    <div
      className="Tasks__StyledTasks-sc-1hy8ads-0 cjVtaH"
    >
      <h2
        className="Title__StyledTitle-sc-11bxhsf-0 frSblw"
      >
        Tarefas
      </h2>
      <button
        className="AddButton__StyledAddButon-ppf0uf-0 gwSZMh"
        onClick={[Function]}
      >
        <span
          className="icon icon-plus Icon__StyledIcon-crr1cg-0 jfRxiR"
          disabled={false}
          fontSize="0.8rem"
          onClick={null}
        />
        Adicionar Tarefa
      </button>
      <div
        className="Task__StyledTask-sc-1fi2258-0 cEDnPQ"
      >
        <span
          className="TaskCheck__StyledTaskCheck-sc-1tgftv-0 kYiGLh"
          onClick={[Function]}
        >
          <button
            className="TaskCheck__StyledButton-sc-1tgftv-1 elQAyk"
          />
          <span
            className="icon icon-check Icon__StyledIcon-crr1cg-0 llBDkg"
            disabled={false}
            fontSize="1.75rem"
            onClick={null}
          />
        </span>
        <div
          className="TaskBody__StyledBody-sc-1p8y2k3-0 cOetdt"
        >
          <textarea
            autoComplete="off"
            className="TaskEdit__StyledTaskEdit-f5yune-0 kXgwgR"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            value="teste 123"
          />
        </div>
        <span
          className="TaskActions__StyledTaskActions-sc-92wkgs-0 fGXkbW"
        >
          <span
            className="icon icon-times Icon__StyledIcon-crr1cg-0 efWoYG"
            disabled={false}
            fontSize="1.5rem"
            onClick={[Function]}
          />
        </span>
      </div>
    </div>
  </section>,
  <main
    className="Container__StyledContainer-sc-1i8buc6-0 SsRsD"
  >
    <div
      className="TravelScriptList__TravelScriptHeader-ty58ph-0 fWwphJ"
    >
      <div />
      <div
        className="AddCollaborators__AddCollaboratorsContainer-sc-9j6qbx-0 jhjoiw"
      >
        <button
          className="StyledInviteButton-sc-1iige8l-0 cJgVMw"
          onClick={[Function]}
        >
          Convidar
        </button>
      </div>
    </div>
    <div
      className="TravelScriptList__TravelMaterialButton-ty58ph-1 gldxts"
    >
      <button
        className="Button__StyledButton-sc-1fc3klx-0 kXhdHl"
        disabled={false}
        fill={true}
        onClick={[Function]}
      >
        Fechar material
      </button>
    </div>
    <div
      className="TravelScriptList__TravelMaterialButton-ty58ph-1 gldxts"
    />
    <div
      className="TravelScriptDay__StyledTravelScriptDay-uatujh-0 faduyg"
    >
      <div
        className="TravelScriptDay__StyledSelects-uatujh-1 dFLRG"
      >
        <div
          className="CitySelect__StyledCitySelect-p43eaz-1 jykjhD"
        >
          <div
            className="CitySelect__StyledDayNumber-p43eaz-0 dnawfo"
          >
            1
          </div>
          <div
            className="Select__StyledSelect-sc-12397s2-0 eBEwLw"
          >
            <div
              className="SelectContainer__StyledSelectContainer-an3t0v-0 jYoLEQ"
            >
              <span
                className="SelectText-eq3jk4-0 iwqxNL"
                onClick={[Function]}
              >
                Selecione uma cidade...
              </span>
              <button
                className="SelectDropdown__StyledSelectDropdown-sc-12m970x-0 RvUCy"
                onClick={[Function]}
                type="button"
              />
            </div>
            <ul
              className="SelectList__StyledSelectList-lfsn73-0 hIsMxM"
            />
          </div>
          <div
            className="CitySelect__StyledLine-p43eaz-2 glAxln"
          />
        </div>
        <div
          className="ServiceSelect__StyledServiceSelect-sc-1rmby0e-0 kwWXAN"
        >
          <div
            className="Select__StyledSelect-sc-12397s2-0 eBEwLw"
          >
            <div
              className="SelectContainer__StyledSelectContainer-an3t0v-0 jYoLEQ"
            >
              <span
                className="SelectText-eq3jk4-0 iwqxNL"
                onClick={[Function]}
              >
                <span
                  className="icon icon-plus Icon__StyledIcon-crr1cg-0 llBDkg"
                  disabled={false}
                  fontSize="1.75rem"
                  onClick={null}
                />
              </span>
              <button
                className="SelectDropdown__StyledSelectDropdown-sc-12m970x-0 RvUCy"
                onClick={[Function]}
                type="button"
              />
            </div>
            <ul
              className="SelectList__StyledSelectList-lfsn73-0 hIsMxM"
            />
          </div>
        </div>
      </div>
      <div
        className="TravelScriptDay__StyledDate-uatujh-2 kemUYJ"
      >
        15/02
      </div>
      <div
        className="TravelScriptService__StyledTravelScript-n3v2ak-1 lhJYVv"
      >
        <span
          className="TravelScriptService__StyledDelIcon-n3v2ak-0 ciWpmh"
        >
          <span
            className="icon icon-times Icon__StyledIcon-crr1cg-0 efWoYG"
            disabled={false}
            fontSize="1.5rem"
            onClick={[Function]}
          />
        </span>
        <div
          className="TravelScriptService__StyledService-n3v2ak-2 cdILFI"
          onClick={[Function]}
        >
          <div
            className="TravelScriptService__StyledBullet-n3v2ak-3 bvzinQ"
            type="RESTAURANT"
          />
          <div
            className="TravelScriptService__StyledServiceInformation-n3v2ak-4 gaWVbb"
          >
            <div
              className="TravelScriptService__StyledType-n3v2ak-6 hGjoUD"
            >
              Restaurante
            </div>
            <span
              className="TravelScriptService__StyledName-n3v2ak-5 kRnSnh"
            >
              Teste
            </span>
          </div>
        </div>
      </div>
    </div>
  </main>,
  <section
    className="Summary__StyledSummary-sc-1mglm3x-0 fKeRKj"
  />,
]
`;
